engine/src/main/java/org/pentaho/di/security/ExitInterceptor.java

```diff
@@ -0,0 +1,34 @@
+/*! ******************************************************************************
+ *
+ * Pentaho
+ *
+ * Copyright (C) 2024 by Hitachi Vantara, LLC : http://www.pentaho.com
+ *
+ * Use of this software is governed by the Business Source License included
+ * in the LICENSE.TXT file.
+ *
+ * Change Date: 2029-07-20
+ ******************************************************************************/
+
+package org.pentaho.di.security;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class ExitInterceptor {
```

Comment:
Add javadoc comments for class explaining this is ire replace the removal of Java17 SecurityManager and to support "unit" test code in PanTest.java and Kitchen.java